platform.py

```diff
@@ -23,6 +23,34 @@ def is_embedded(self):
         return True
 
     def configure_default_packages(self, variables, targets):
+        # configure arduino core package.
+        # select the right one based on the build.core, disable other one.
+        board = variables.get("board")
+        board_config = self.board_config(board)
+        build_core = variables.get(
+            "board_build.core", board_config.get("build.core", "arduino"))
+
+        frameworks = variables.get("pioframework", [])
+        if "arduino" in frameworks:
+            if build_core == "arduino":
+                self.frameworks["arduino"]["package"] = "framework-arduino-mbed"
+                self.packages["framework-arduinopico"]["optional"] = True
+                self.packages["toolchain-rp2040-earlephilhower"]["optional"] = True 
+                self.packages.pop("toolchain-rp2040-earlephilhower", None)
+            elif build_core == "earlephilhower":
+                self.frameworks["arduino"]["package"] = "framework-arduinopico"
+                self.packages["framework-arduino-mbed"]["optional"] = True
+                self.packages.pop("toolchain-gccarmnoneeabi", None)
+                self.packages["toolchain-rp2040-earlephilhower"]["optional"] = False                
+            else:
+                sys.stderr.write(
+                    "Error! Unknown build.core value '%s'. Don't know which Arduino core package to use." % build_core)
+                env.Exit(1)
+
+        # if we want to build a filesystem, we need the tools.
+        if "buildfs" in targets:
+            self.packages['tool-mklittlefs-rp2040-earlephilhower']['optional'] = False
```

Comment:
Please use `"` just to be consistent with other code.

Reply:
Agreed, will fix.

Reply:
Is equalized.